### PR TITLE
Prototype/fdb layer ebtree views

### PR DESCRIPTION
**Pull request by davisp**
## Overview

Full support for reduce functions using ebtree for all map/reduce data.

## Testing recommendations

`make check`

## Related Issues or Pull Requests

This is an alternative approach to:

#2984 
#3018

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [x] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/couch_views/src/couch_views_fdb.erl [hunk 1]
```diff
@@ -244,200 +373,223 @@ clear_index(Db, Signature) ->
...
+    min_order(config:get_integer("couch_views", "id_btree_node_size", 100));
+get_order(view_btree) ->
+    min_order(config:get_integer("couch_views", "view_btree_node_size", 100)).
+
+
+min_order(V) when is_integer(V), V < 2 ->
+    2;
+min_order(V) when is_integer(V), V rem 2 == 0 ->
+    V;
+min_order(V) ->
```
Comment (rnewson) [inline src/couch_views/src/couch_views_fdb.erl:440]: `when is_integer(V) ->`

File: src/couch_views/src/couch_views_fdb.erl [hunk 2]
```diff
@@ -244,200 +373,223 @@ clear_index(Db, Signature) ->
...
 
-    % Track a database level rollup for calls to
-    % GET /dbname
-    DbKey = db_kv_size_key(DbPrefix),
-    erlfdb:add(Tx, DbKey, Increment).
+persist_chunks(Tx, set, [Key, Value]) ->
+    Chunks = fabric2_fdb:chunkify_binary(Value),
+    lists:foldl(fun(Chunk, Id) ->
+        ChunkKey = erlfdb_tuple:pack({Id}, Key),
+        erlfdb:set(Tx, ChunkKey, Chunk),
```
Comment (rnewson) [inline src/couch_views/src/couch_views_fdb.erl:475]: aegis:encrypt here

Comment (davisp) [reply src/couch_views/src/couch_views_fdb.erl:475]: That should be part of the encode function. Need to add that as well.

File: src/mango/src/mango_idx_view.erl [hunk 3]
```diff
@@ -34,6 +34,7 @@
 -include_lib("couch/include/couch_db.hrl").
 -include("mango.hrl").
 -include("mango_idx.hrl").
+-include("mango_idx_view.hrl").
```
Comment (garrensmith) [inline src/mango/src/mango_idx_view.erl:37]: I've only had a very quick look at this PR. But I'm not sure why the mango indexes are affected? From our discussion, we agreed that map only indexes wouldn't use ebtree?

File: src/couch_views/src/couch_views_fdb.erl [hunk 4]
```diff
@@ -126,92 +129,218 @@ set_update_seq(TxDb, Sig, Seq) ->
...
 
-get_row_count(TxDb, #mrst{sig = Sig}, ViewId) ->
+set_trees(TxDb, Mrst) ->
+    #mrst{
+        sig = Sig,
+        language = Lang,
+        views = Views
+    } = Mrst,
+    Mrst#mrst{
+        id_btree = open_id_tree(TxDb, Sig),
```
Comment (garrensmith) [inline src/couch_views/src/couch_views_fdb.erl:139]: Why do we need an id tree?

**Comment by davisp** [comment]
Closing in favor of #3164

**Pull request closed without merging**